File: app/src/main/java/antoninovitale/dropcodechallenge/list/BeerListPresenter.java
package antoninovitale.dropcodechallenge.list;

import java.util.List;

import antoninovitale.dropcodechallenge.api.model.Beer;
import antoninovitale.dropcodechallenge.viewmodel.CurrentStatus;

/**
 * Created by antoninovitale on 02/09/2017.
 */
public interface BeerListPresenter {
    void onFloatingButtonClick();

    void onRefresh();

    void onChanged(List<Beer> beers);

    void onListItemClick(int position);

    void onChanged(CurrentStatus currentStatus);
}

File: app/src/main/java/antoninovitale/punkapi/app/list/BeerListPresenter.kt
package antoninovitale.punkapi.app.list

import antoninovitale.punkapi.app.api.model.Beer
import antoninovitale.punkapi.app.viewmodel.CurrentStatus

/**
 * Created by antoninovitale on 02/09/2017.
 */
interface BeerListPresenter {
    fun onRefresh()
    fun onChanged(beers: List<Beer?>?)
    fun onListItemClick(position: Int)
    fun onChanged(currentStatus: CurrentStatus?)
}